### ISSUE TITLE
mission objective in templates ?

### ISSUE THREAD

**Issue by shirishag75** [mars-sim/mars-sim#268]
Are there any mission objective templates or there isn't ? If there is, how does one view to know objectives of the template to have a win scenario. 

@mokun @Urwumpe @larsnaesbye 

It would be nice to have each template for e.g. template:1 to have a scenario so we know what objectives need to be achieved. Right now, the commander/gamer has no idea what the objectives are. Is there a proposal to -

a. Code the objectives and share it in the headless version so I/ the gamer knows. 
b. Code the progress on the objectives and any sub-objectives which show what needs to be done in order to be done. 

For e.g. an objective would be to build tow things in template 1 - 

a. Some additional beds. 
b. a hydroponics garden to grow food and such with sensors and all. 

template 2 could have start on having an EVA specialized suit room for more EVA suits, development and repair 

it would be sort of story mode or also endless sandbox mode but dependant upon the objectives being met. The sandbox mode should be open only after certain objectives are met.

**Comment by Urwumpe** [comment]
I think we don't have anything like that yet. Any wishes how it should be like? 

I personally would prefer using script files to define such scenarios and objectives. Allows more flexibility and would be easier to extend later. A pure data format like XML would be tougher there, but maybe easier to use.

**Comment by shirishag75** [comment]
> I think we don't have anything like that yet. Any wishes how it should be like?
> 
> I personally would prefer using script files to define such scenarios and objectives. Allows more flexibility and would be easier to extend later. A pure data format like XML would be tougher there, but maybe easier to use.

While I would prefer .json if that is possible but that's upto the implementors and how they look to implement it. The idea and I have shared before is to create two things - 

a. Have a winnable scenario 
b. Also grow skills. I know we do have skils but there is nothing to know at what level your skills start at the beginning. 

I have number of scenarios or objectives at times but objectives should have some of **conditions met** kind of thing. For e.g. having a morgue, having a nursery station, having medical lab, having a school and so on and so forth. 

There is another thing I was thinking about. They should be some cap for settlements, maybe 10-15 and once that is done/reached maybe have satellite settlements and you can have trading between them with settlements trading goods with each other or trading goods with knowledge or something like that, but that probably be somewhat far down the line. 

For e.g. I was reading somewhere that the sand/regolith on Mars is much more finer than on Earth and more toxic in nature. There have been some alternatives suggested to NASA and each has its own pros and cons. I don't know how much of that has been discussed or what has been assumed or presumed therein. 

https://www.mars-one.com/faq/mission-to-mars/what-are-the-risks-of-dust-and-sand-on-mars
https://en.wikipedia.org/wiki/Martian_soil

**Comment by Urwumpe** [comment]
So, JSON format for any template. Can be done.

Which kind of mission objective do you need for initial capability? "Build n of X"? "Locate mineral site with least 50% kaolinite?" 

The population caps should go into a new issue, that is an slightly unrelated feature.

**Comment by shirishag75** [comment]
Reply in-line -

> So, JSON format for any template. Can be done.
> 

cool. 

> Which kind of mission objective do you need for initial capability? "Build n of X"? "Locate mineral site with least 50% kaolinite?"
> 

I actually have no idea. In the headless package I prefer to use, the gamer has no idea about the surroundings and what is urgent or what needs to be priortized. As of right now, it seems, it seems we have only 2-3 things,  people eating, the gardenbots doing gardening, supply missions but no clear objectives as to why we are doing that. 

Two objectives are clear to me so far. As this is supposed to be a five year program, at least initially, the idea is to have an initial settlement which can have 10-15 members in the settlement. The initial count is 4-5. 

So when they are dropped (presumably by some rocket) , you need to have quite a few things to survive - 

a. Enough oxygen, co2 and whatever else is needed in order to survive. 
b. Enough food so things can work for 5 years.  
c. Enough power for 24x7 
d. Technology and constantly figuring and fixing things which are broken. 
e. Communication - between mars and an orbiter (assuming or presuming) and orbiter and earth. 
f. Climate things (although in the beginning this might be done by the orbitor) . 

The big question is how to have goals, how to present it to user and how to reward the user/gamer/commander when they have reached that goal. Maybe give them choices in choice of rewards so depending on what might be needed/required they will be given. 

When you meet conditions to win, you get a choice and you get the resources or technology which is needed. 

> The population caps should go into a new issue, that is an slightly unrelated feature.

agreed, at least we can talk on the above. I would file a new ticket to have population caps at least initially.

**Comment by mokun** [comment]
@shirishag75  @Urwumpe 

Let's first figure out how to proceed with saving and load JSON data set.  

After that, we'll have a lot of flexibility to play with.

It would be cool for players to design their own custom mission templates in future for sure.

**Comment by mokun** [comment]
@bevans2000 

Guess we can create a list of these small objectives for player to achieve.

Most strategy games use them to guide players to familiarize with what they should do at the beginning. 

No only these objectives can be made into a part of a tutorial but also  a part of a SCENARIO that you've been working on in https://github.com/mars-sim/mars-sim/issues/412
